### PR TITLE
bpo-44252: Correctly implement gc support for SSLError objects

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue44252

### CODE REVIEW

File: Lib/test/test_ssl.py [hunk 1]
```diff
@@ -1824,6 +1824,14 @@ def test_bad_server_hostname(self):
             ctx.wrap_bio(ssl.MemoryBIO(), ssl.MemoryBIO(),
                          server_hostname="example.org\x00evil.com")
 
+    def test_gc_support(self):
```
Comment (vstinner) [inline Lib/test/test_ssl.py]: Please add a comment mentioning bpo-44252 to explain why you added this test. usually, we don't test the GC protocol explicitly on types.

**Review by vstinner — APPROVED** [review]
LGTM. I'm not sure that the unit test is useful. If you want to keep it, maybe add a comment.

Using the following change, I can reproduce https://bugs.python.org/issue44252

```
diff --git a/Lib/site.py b/Lib/site.py
index 939893eb5e..c02987e220 100644
--- a/Lib/site.py
+++ b/Lib/site.py
@@ -672,3 +672,5 @@ def exists(path):
 
 if __name__ == '__main__':
     _script()
+
+import gc; gc.set_threshold(5)
```

Without this PR:

```
$ ./python -m test test_ssl -m test_ssl_cert_verify_error -v 
...
Fatal Python error: Segmentation fault

Current thread 0x00007f25b0858640 (most recent call first):
  File "/home/vstinner/python/3.10/Lib/traceback.py", line 132 in format_exception
  File "/home/vstinner/python/3.10/Lib/test/test_ssl.py", line 262 in handle_error
  File "/home/vstinner/python/3.10/Lib/test/test_ssl.py", line 2401 in wrap_conn
...
```

With this PR:

```
$ ./python -m test test_ssl -v
(...)
Tests result: SUCCESS
```

File: Lib/test/test_ssl.py (hunk 1 above)
Comment (pablogsal) [reply Lib/test/test_ssl.py]: Ok, but I don't think is bad to add a test that checks that we have gc support for a given type properly

Comment (pablogsal) [reply Lib/test/test_ssl.py]: I am actually going to remove the test

**Review by vstinner — APPROVED** [review]
LGTM.

**Comment by erlend-aasland** [comment]
Thanks for fixing this, Pablo! Sorry for introducing the regression.

**Pull request merged**